server/src/main/java/org/apache/druid/server/initialization/jetty/JettyServerModule.java

```diff
@@ -483,6 +505,28 @@ public boolean isShowServlet()
           return false;
         }
 
+
+        @Override
+        public void generateAcceptableResponse(
+            Request baseRequest,
+            HttpServletRequest request,
+            HttpServletResponse response,
+            int code,
+            String message,
+            String contentType
+        ) throws IOException
+        {
+          if (config.isDefaultJettyErrorResponseInJson()) {
+            super.generateAcceptableResponse(baseRequest, request, response, code, message,
+                                             MimeTypes.Type.APPLICATION_JSON.asString()
+            );
+          } else {
+            super.generateAcceptableResponse(baseRequest, request, response, code, message,
```

Comment:
Why are we setting to html here? If this new config is not set, we should still depend on the Accept header to determine the response format right?